test/smartwallet/smartWallet.test.ts

```diff
@@ -1,943 +1,537 @@
-//         );
-//       });
-//     });
-//   }
-// );
-
-// /* TODO - ADD tests for recover method */
+import { ethers as hardhat } from 'hardhat';
+import { SignerWithAddress } from "@nomiclabs/hardhat-ethers/signers";
+import chai, { expect} from 'chai';
+import { FakeContract, smock } from '@defi-wonderland/smock';
+import chaiAsPromised from 'chai-as-promised';
+import { loadFixture } from '@nomicfoundation/hardhat-network-helpers';
+import { TypedDataUtils } from '@metamask/eth-sig-util';
+import {
+    getLocalEip712Signature,
+    TypedRequestData,
+    RelayRequest,
+    ForwardRequest,
+    RelayData} from '../utils/EIP712Utils';
+import { SignTypedDataVersion } from '@metamask/eth-sig-util';
+import { Wallet } from 'ethers';
+import { SmartWallet } from 'typechain-types';
+import { BaseProvider } from '@ethersproject/providers';
+
+chai.use(smock.matchers);
+chai.use(chaiAsPromised);
+
+const ZERO_ADDRESS = hardhat.constants.AddressZero;
+const ONE_FIELD_IN_BYTES = 32;
+
+describe('SmartWallet', function(){
+    function createRequest(
+        request: Partial<ForwardRequest>,
+        relayData: Partial<RelayData>
+    ): RelayRequest {
+        const baseRequest: RelayRequest = {
+            request:{
+                relayHub: ZERO_ADDRESS,
+                from: ZERO_ADDRESS,
+                to: ZERO_ADDRESS,
+                tokenContract: ZERO_ADDRESS,
+                value: '0',
+                gas: '10000',
+                nonce: '0',
+                tokenAmount: '0',
+                tokenGas: '50000',
+                data: '0x'
+            } as ForwardRequest,            
+            relayData:{
+                gasPrice: '1',
+                relayWorker: ZERO_ADDRESS,
+                callForwarder: ZERO_ADDRESS,
+                callVerifier: ZERO_ADDRESS
+            } as RelayData
+        } as RelayRequest
+
+        return {
+            request: {
+                ...baseRequest.request,
+                ...request,
+            },
+            relayData: {
+                ...baseRequest.relayData,
+                ...relayData,
+            }
+        }
+    }
+    
+    function getSuffixData(typedRequestData: TypedRequestData):string{
+        const encoded =TypedDataUtils.encodeData(
+            typedRequestData.primaryType,
+            typedRequestData.message,
+            typedRequestData.types,
+            SignTypedDataVersion.V4
+        );
+
+        const messageSize = Object.keys(typedRequestData.message).length;
+
+        return '0x'+(encoded.slice(messageSize * ONE_FIELD_IN_BYTES)).toString('hex');
+    }
+
+    describe('Function initialize()', function(){
+        let fakeToken:FakeContract;
+
+        async function prepareFixture(){
+            const smartWalletFactory =await hardhat.getContractFactory('SmartWallet');
+            const smartWallet = await smartWalletFactory.deploy();
+            const [owner, worker, utilSigner] = await hardhat.getSigners();
```

Comment:
The signers does not change during the executions, it should not be inside a fixture.

Reply:
Done.